src/test/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/JwtServiceTest.kt

```diff
@@ -39,9 +43,21 @@ class JwtServiceTest {
     assertThat(jwtService.authorities(jwt)).containsExactly("ROLE_SLM_CREATE_BARCODE")
   }
 
+  @Test
+  fun `the generated JWT should expire at midnight on the day of expiry`() {
+    val jwtService = jwtService(
+      expiry = Duration.of(1, ChronoUnit.DAYS),
+      clock = Clock.fixed(Instant.parse("2021-11-26T12:18:05Z"), ZoneId.of("Europe/London"))
+    )
+    val jwt = jwtService.generateToken("some.email@company.com")
+
+    val expiresAt = jwtService.expiresAt(jwt)
+    assertThat(expiresAt).isEqualTo(Instant.parse("2021-11-28T00:00:00Z"))
```

Comment:
Yeah, much nicer with a fixed clock. Not only is it not duplicating the production code, but its absolutely clear what the expectation is 👍